src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Common.Core;
+
+namespace Microsoft.R.Host.Broker.RemoteUri {
+    public class RemoteUriHelper {
+        public static async Task HandlerAsync(HttpContext context) {
+            var url = context.Request.Headers["x-rtvs-url"];
+
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            request.Method = context.Request.Method;
+            request.UseDefaultCredentials = true;
+            SetRequestHeaders(request, context.Request.Headers);
+
+            if (context.Request.ContentLength > 0) {
+                Stream reqStream = await request.GetRequestStreamAsync();
+                await context.Request.Body.CopyToAsync(reqStream);
+                await reqStream.FlushAsync();
+            }
+
+            HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync();
+
+            SetResponseHeaders(response, context.Response);
+
+            Stream respStream = response.GetResponseStream();
+            await respStream.CopyToAsync(context.Response.Body);
+            await context.Response.Body.FlushAsync();
+        }
+
+        private static long GetLong(string value) {
+            long ret;
+            if (long.TryParse(value, out ret)) {
+                return ret;
+            }
+            return 0;
+        }
+
+        private static DateTime GetDateTime(string value) {
+            DateTime ret;
+            if (DateTime.TryParse(value, out ret)) {
+                return ret;
+            }
+            return new DateTime();
+        }
+
+        private static void SetRequestHeaders(HttpWebRequest request, IHeaderDictionary requestHeaders) {
+            // copy headers to avoid messing with original request headers
+            Dictionary<string, string> headers = new Dictionary<string, string>();
+            foreach (var pair in requestHeaders) {
+                headers.Add(pair.Key, pair.Value);
+            }
+
+            if (headers.ContainsKey("Accept")) {
```

Comment:
I'm kinda surprised they didn't just wire the properties directly to the corresponding headers. Oh well... might be worth revisiting in the future to see if there's a way to go even lower in level and avoid those helpful request objects that seem to hinder more than help in this case. For now, though, let's stick with this.